sdk/eventgrid/azure-messaging-eventgrid/migration-guide.md

```diff
@@ -0,0 +1,141 @@
+# Guide for migrating to `azure-messaging-eventgrid` from `microsoft-azure-eventgrid`
+
+This guide is intended to assist in the migration to `azure-messaging-eventgrid` from `microsoft-azure-eventgrid`. It will focus on side-by-side comparisons for similar operations between the two packages.
+
+We assume that you are familiar with the old SDK `microsoft-azure-eventgrid`. If not, please refer to the new SDK README for [azure-messaging-eventgrid][README] directly rather than this migration guide.
+
+## Table of contents
+
+- [Guide for migrating to `azure-messaging-eventgrid` from `microsoft-azure-eventgrid`](#guide-for-migrating-to-azure-messaging-eventgrid-from-microsoft-azure-eventgrid)
+    - [Table of contents](#table-of-contents)
+    - [Migration benefits](#migration-benefits)
+        - [Cross Service SDK improvements](#cross-service-sdk-improvements)
+    - [Important changes](#important-changes)
+        - [Group id, artifact id, and package names](#group-id-artifact-id-and-package-names)
+        - [Instantiating clients](#instantiating-clients)
+        - [Send events to an Event Grid Topic](#send-events-to-an-event-grid-topic)
+        - [Deserialize events and data](#deserialize-events-and-data)
+    - [Additional samples](#additional-samples)
+
+## Migration benefits
+
+A natural question to ask when considering whether or not to adopt a new version or library is its benefits. As Azure has matured and been embraced by a more diverse group of developers, we have been focused on learning the patterns and practices to best support developer productivity and to understand the gaps that the Java
+client libraries have.
+
+There were several areas of consistent feedback expressed across the Azure client library ecosystem. The most important is that the client libraries for different Azure services have not had a consistent organization, naming, and API structure. Additionally, many developers have felt that the learning curve was difficult, and the APIs did not offer a good, approachable, and consistent onboarding story for those learning Azure or exploring a specific Azure service.
+
+To improve the development experience across Azure services, a set of uniform [design guidelines][Guidelines] was created for all languages to drive a
+consistent experience with established API patterns for all services. A set of [Java design guidelines][GuidelinesJava] was introduced to ensure that Java clients have a natural and idiomatic feel with respect to the Java ecosystem. Further details are available in the guidelines
+for those interested.
+
+### Cross Service SDK improvements
+
+The modern Event Hubs client library also provides the ability to share in some of the cross-service improvements made to the Azure development experience, such as
+
+- Using the new azure-identity library to share a single authentication approach between clients.
```

Comment:
I don't think Event Grid uses `azure-identity`. We only support key credential and sas credential and neither of them need azure-identity.

Reply:
Yes, they're in `azure-core`. I removed it since we don't use identity yet.